src/kanata/output_logic/zippychord.rs

```diff
@@ -0,0 +1,372 @@
+    /// activation is no longer possible.
+    zchd_prioritized_chords: Option<Arc<parking_lot::Mutex<ZchPossibleChords>>>,
+    /// Tracks the previous output character count
+    /// because it may need to be erased (see `zchd_prioritized_chords).
+    zchd_previous_activation_output_count: u16,
+    /// In case of output being empty for interim chord activations, this tracks the number of
+    /// characters that need to be erased.
+    zchd_characters_to_delete_on_next_activation: u16,
+    /// Tracker for time until previous state change to know if potential stale data should be
+    /// cleared. This is a contingency in case of bugs or weirdness with OS interactions, e.g.
+    /// Windows lock screen weirdness.
+    ///
+    /// This counts upwards to a "reset state" number.
+    zchd_ticks_since_state_change: u16,
+    /// Zch has a time delay between being disabled->pending-enabled->truly-enabled to mitigate
+    /// against unintended activations. This counts downwards from a configured number until 0, and
+    /// at 0 the state transitions from pending-enabled to truly-enabled if applicable.
+    zchd_ticks_until_enabled: u16,
+    /// Current state of caps-word, which is a factor in handling capitalization.
+    zchd_is_caps_word_active: bool,
+    /// Current state of lsft which is a factor in handling capitalization.
+    zchd_is_lsft_active: bool,
+    /// Current state of rsft which is a factor in handling capitalization.
+    zchd_is_rsft_active: bool,
+}
+
+impl ZchDynamicState {
+    fn zchd_is_disabled(&self) -> bool {
+        self.zchd_enabled_state == ZchEnabledState::Disabled
+    }
+    fn zchd_tick(&mut self, is_caps_word_active: bool) {
+        const TICKS_UNTIL_FORCE_STATE_RESET: u16 = 10000;
+        self.zchd_ticks_since_state_change += 1;
+        self.zchd_is_caps_word_active = is_caps_word_active;
+        if self.zchd_enabled_state == ZchEnabledState::WaitEnable {
+            self.zchd_ticks_until_enabled = self.zchd_ticks_until_enabled.saturating_sub(1);
+            if self.zchd_ticks_until_enabled == 0 {
+                self.zchd_enabled_state = ZchEnabledState::Enabled;
+            }
+        }
+        if self.zchd_ticks_since_state_change > TICKS_UNTIL_FORCE_STATE_RESET {
+            self.zchd_reset();
+        }
+    }
+    fn zchd_state_change(&mut self, cfg: &ZchConfig) {
+        self.zchd_ticks_since_state_change = 0;
+        self.zchd_ticks_until_enabled = cfg.zch_cfg_ticks_wait_enable;
+    }
+
+    /// Clean up the state.
+    fn zchd_reset(&mut self) {
+        log::debug!("zchd reset state");
+        self.zchd_enabled_state = ZchEnabledState::Enabled;
+        self.zchd_is_caps_word_active = false;
+        self.zchd_is_lsft_active = false;
+        self.zchd_is_rsft_active = false;
+        self.zchd_input_keys.zchik_clear();
+        self.zchd_prioritized_chords = None;
+        self.zchd_previous_activation_output_count = 0;
```

Comment:
I'm just trying out the following change. Will let you know if it works.
```suggestion
        self.zchd_previous_activation_output_count = 0;
        self.zchd_ticks_since_state_change = 0;
```

Reply:
That seems to work. I didn't have another crash since this change.